### ISSUE TITLE
Merge script use of origin and upstream

### ISSUE THREAD

**Issue by jfehrle** [coq/coq#10625]
`MERGING.md` says:

```
To merge the PR proceed in the following way

$ git checkout master
$ git pull
$ dev/tools/merge-pr.sh XXXX
$ git push upstream

where `XXXX` is the number of the PR to be merged and `upstream` is the name
of your remote pointing to `git@github.com:coq/coq.git`.
```

But the script requires that `origin` points to master (below), so what is the point of referring to `upstream` in the instructions?  When would you "Continue anyway"?

I can put in PR to update the instructions.

```
error: remote origin does not point to the official Coq repo
error: that is git@github.com:coq/coq
error: it points to https://github.com/jfehrle/coq.git instead
Continue anyway? [y/N] n
```

**Comment by jfehrle** [comment]
Also, seems that the second line of the error messages should refer to "https://github.com/coq/coq.git" for consistency with "git remote -v".

**Comment by Zimmi48** [comment]
Nope the script does not actually require anything regarding `origin`. After ensuring that `git remote -v` tells you that `upstream` does indeed point to `git@github.com:coq/coq` or `https://github.com/coq/coq` possibly with the `.git` suffix, you should do `git branch master --set-upstream-to upstream/master`. Then the merging script shouldn't complain like that anymore. And `git pull` will work without any additional argument.

**Comment by jfehrle** [comment]
> Nope the script does not actually require anything regarding `origin`.

Huh?  Then why the message?

What is the conventional setup?  At the moment I have:

```
jfehrle@LAPTOP-LB5T3R83:/mnt/c/proj/coq2$ git remote -v
jfehrle https://github.com/jfehrle/coq.git (fetch)
jfehrle https://github.com/jfehrle/coq.git (push)
origin  https://github.com/coq/coq.git (fetch)
origin  https://github.com/coq/coq.git (push)
```

Before this morning, I had "origin" pointing to jfehrle/coq.git and "upstream" pointing to coq/coq.git.

**Comment by Zimmi48** [comment]
The conventional setup is the one you had before this morning, also documented in https://github.com/coq/coq/blob/master/CONTRIBUTING.md#git-documentation-tips-and-tricks. However, to ensure the conventional setup is complete you need to ensure that `master` is set to track `upstream/master`. You could also keep the current setup, and in this case you would need to ensure that the `master` branch tracks `origin/master`. The issue and the reason why you got the error message with the previous setup is that your `master` branch tracked `origin/master` but `origin` pointed to your fork. That is why the error message talks about `origin`. If your fork remote had been `abcdefg`, the error message would have talked about `abcdefg`. That's also why my recommended fix is to use `git branch --set-upstream-to`.

**Comment by jfehrle** [comment]
Not working; I tried several variations.  Any ideas?

```
jfehrle@LAPTOP-LB5T3R83:/mnt/c/proj/coq$ git remote -v
origin  https://github.com/jfehrle/coq.git (fetch)
origin  https://github.com/jfehrle/coq.git (push)
upstream        https://github.com/coq/coq.git (fetch)
upstream        https://github.com/coq/coq.git (push)

jfehrle@LAPTOP-LB5T3R83:/mnt/c/proj/coq$ git branch master --set-upstream-to upstream/master
error: the requested upstream branch 'upstream/master' does not exist
hint:
hint: If you are planning on basing your work on an upstream
hint: branch that already exists at the remote, you may need to
hint: run "git fetch" to retrieve it.
hint:
hint: If you are planning to push out a new local branch that
hint: will track its remote counterpart, you may want to use
hint: "git push -u" to set the upstream config as you push.
```

**Comment by jfehrle** [comment]
If I ignore `error: remote origin does not point to the official Coq repo` message will the script still work (more of less)?

**Comment by Zimmi48** [comment]
Given the error message given by `git branch master --set-upstream-to upstream/master`, I suggest you run this other command before `git fetch upstream`.

**Comment by Zimmi48** [comment]
>If I ignore error: remote origin does not point to the official Coq repo message will the script still work (more of less)?

Yes, I think so, but it will also mean that the script won't check that your branch is up-to-date before merging.

**Comment by jfehrle** [comment]
> I suggest you run this other command before `git fetch upstream`.

Which other command?

**Comment by Zimmi48** [comment]
I suggest you run `git fetch upstream` before `git branch master --set-upstream-to upstream/master`.

**Comment by jfehrle** [comment]
Looks good, thanks.
